packages/webapp/pages/plus/index.tsx

```diff
@@ -17,21 +24,69 @@ const PlusDesktop = dynamic(() =>
   ).then((mod) => mod.PlusDesktop),
 );
 
-const PlusPage = (): ReactElement => {
+type PlusPageProps = Pick<GiftUserContextData, 'giftToUser'>;
+
+const seo: NextSeoProps = {
+  title: getTemplatedTitle('Unlock Premium Developer Features with Plus'),
+  openGraph: { ...defaultOpenGraph },
+  description:
+    'Upgrade to daily.dev Plus for an ad-free experience, custom feeds, bookmark folders, clickbait shield, and more.',
+};
+
+const PlusPage = ({ giftToUser }: PlusPageProps): ReactElement => {
+  const [userToGift, setUserToGift] = useState(giftToUser);
   const { isReady } = useRouter();
   const isLaptop = useViewSize(ViewSize.Laptop);
+  const pageProps: GiftUserContextData = {
+    giftToUser: userToGift,
+    onUserChange: setUserToGift,
+  };
 
   if (!isReady) {
     return null;
   }
 
   if (isLaptop) {
-    return <PlusDesktop />;
+    return (
+      <GiftUserContext.Provider value={pageProps}>
+        <PlusDesktop />
+      </GiftUserContext.Provider>
+    );
   }
 
-  return <PlusMobile />;
+  return (
+    <GiftUserContext.Provider value={pageProps}>
+      <PlusMobile />
+    </GiftUserContext.Provider>
+  );
```

Comment:
Maybe do to avoid repeating and passing values to provider:
````
<GiftUserContext.Provider>
    {isLaptop ? <PlusDesktop /> : <PlusMobile />
</GiftUserContext.Provider>